concordances/handlers.go

```diff
@@ -0,0 +1,257 @@
+	log.Info("Registering admin handlers")
+	healthService := newHealthService(config)
+	hc := health.HealthCheck{SystemCode: h.conf.AppSystemCode, Name: h.conf.AppName, Description: "Stores concordances in cache and notifies downstream services", Checks: healthService.checks}
+	router.HandleFunc(healthPath, health.Handler(hc))
+	router.HandleFunc(status.PingPath, status.PingHandler)
+	router.HandleFunc(status.GTGPath, status.NewGoodToGoHandler(healthService.gtgCheck))
+	router.HandleFunc(status.BuildInfoPath, status.BuildInfoHandler)
+
+	monitoringRouter := httphandlers.TransactionAwareRequestLoggingHandler(log.StandardLogger(), router)
+	monitoringRouter = httphandlers.HTTPMetricsHandler(metrics.DefaultRegistry, monitoringRouter)
+	http.Handle("/", monitoringRouter)
+}
+
+func (h *ConcordancesRwHandler) HandleGet(rw http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	//400
+	if h.invalidPath(vars) {
+		h.HandleBadRequest(rw, r)
+		return
+	}
+	uuid := vars[UUID_Param]
+	model, err := h.srv.Read(uuid)
+
+	//503
+	if err != nil {
+		h.handleServiceUnavailable("retrieving", err.Error(), rw, r)
+		return
+	}
+	//404
+	if model.ConcordedIds == nil {
+		h.handleNotFound(uuid, rw, r)
+		return
+	}
+
+	output, err := json.Marshal(&model)
+	//404
+	if err != nil {
+		h.handleNotFound(uuid, rw, r)
+		return
+	}
+	//200
+	rw.Header().Set("Content-Type", ContentTypeJson)
+	rw.WriteHeader(http.StatusOK)
+	rw.Write(output)
+}
+
+func (h *ConcordancesRwHandler) HandlePut(rw http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	//400
+	if h.invalidPath(vars) {
+		h.HandleBadRequest(rw, r)
+		return
+	}
+	uuid := vars[UUID_Param]
+	if r.ContentLength <= 0 {
+		h.handleBadPayload(ErrorMsg_BadJson, rw, r)
+		return
+	}
+	body, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+	//400
+	if err != nil {
+		h.handleBadPayload(ErrorMsg_BadJson, rw, r)
+		return
+	}
+
+	model := db.Model{}
+	err = json.Unmarshal(body, &model)
+	//400
+	if err != nil {
+		h.handleBadPayload(ErrorMsg_BadJson, rw, r)
+		return
+	}
+
+	err = h.invalidModel(uuid, model)
+	//400
+	if err != nil {
+		h.handleBadPayload(err.Error(), rw, r)
+		return
+	}
+
+	created, err := h.srv.Write(model)
+
+	//503
+	if err != nil {
+		h.handleServiceUnavailable("storing", err.Error(), rw, r)
+		return
+	}
+
+	if created {
+		rw.WriteHeader(http.StatusCreated)
+		return
+	} else {
+		rw.WriteHeader(http.StatusOK)
+	}
+}
+
+func (h *ConcordancesRwHandler) HandleDelete(rw http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	//400
+	if h.invalidPath(vars) {
+		h.HandleBadRequest(rw, r)
+		return
+	}
+	uuid := vars[UUID_Param]
+	deleted, err := h.srv.Delete(uuid)
+	//503
+	if err != nil {
+		h.handleServiceUnavailable("deleting", err.Error(), rw, r)
+		return
+	}
+	//404
+	if !deleted {
+		h.handleNotFound(uuid, rw, r)
+		return
+	}
+	//204
+	rw.WriteHeader(http.StatusNoContent)
+}
+
+func (h *ConcordancesRwHandler) HandleCount(rw http.ResponseWriter, r *http.Request) {
```

Comment:
If we're not implementing Count in the Dynamo service, then we shouldn't be handling it with a success.  It'll always return 0 which will be wrong.